### PR TITLE
5.3 done

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
+
+            while (bufferedReader.ready()) {
+                readText.append((char) bufferedReader.read());
+            }
+
+            for (String word : readText
+                    .toString()
+                    .toLowerCase()
+                    .split("\\W+")) {
+                if (word.startsWith("w")) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:33]: please make `w` a constant field

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+                    result.add(word);
+                }
+            }
+
+            Collections.sort(result);
+            return result.toArray(new String[result.size()]);
+
+        } catch (IOException exception) {
+            System.out.println("ex");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:42]: rethrow this as a RuntimeException

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+        }
         return null;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:44]: after rethrowing Exception you should remove this line

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -39,8 +42,7 @@
             return result.toArray(new String[result.size()]);
 
         } catch (IOException exception) {
-            System.out.println("ex");
+            throw new RuntimeException("Fail is not read");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:45]:
```suggestion
            throw new RuntimeException("File cannot be read");
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -30,7 +33,7 @@
                     .toString()
                     .toLowerCase()
                     .split("\\W+")) {
-                if (word.startsWith("w")) {
+                if (word.startsWith(String.valueOf(symbolThatAllWordsStartWith))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:36]: why is `symbolThatAllWordsStartWith` a char when we need a String here?

File: src/main/java/core/basesyntax/FileWork.java [hunk 6]
```diff
@@ -16,6 +16,9 @@
  */
 
 public class FileWork {
+
+    private final char symbolThatAllWordsStartWith = 'w';
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:20]: this is a constant
what are checkstyle requirements for constants? should they be static or not? should names be written in lower or upper case?

File: src/main/java/core/basesyntax/FileWork.java [hunk 7]
```diff
@@ -1,14 +1,48 @@
...
+                    .toLowerCase()
+                    .split("\\W+")) {
+                if (word.startsWith(SYMBOL_ALL_WORDS_STARTS_WITH)) {
+                    result.add(word);
+                }
+            }
+
+            Collections.sort(result);
+            return result.toArray(new String[result.size()]);
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:43]:
```suggestion
            }
            Collections.sort(result);
            return result.toArray(new String[result.size()]);
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 8]
```diff
@@ -1,14 +1,48 @@
...
+        } catch (IOException exception) {
+            throw new RuntimeException("File cannot be read");
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:45]: let's add exception object to constructor too in order to keep all information about the original exception

File: src/main/java/core/basesyntax/FileWork.java [hunk 9]
```diff
@@ -1,14 +1,48 @@
...
 /**
  * <p>Дано файл, потрібно прочитати його вміст і вибрати всі слова що починаються на `w`.
  * Результат повернути у вигляді відсортованого масиву (за замовчуванням). Всі слова повинні
  * бути в нижньому регістрі. У випадку якщо таких слів не знайдено повернути пустий масив.
  * Приклад: Width world Wide web
  * Результат: web wide width world</p>
  */
+
 public class FileWork {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:19]:
```suggestion
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 10]
```diff
@@ -1,14 +1,45 @@
...
+                    .toString()
+                    .toLowerCase()
+                    .split("\\W+")) {
+                if (word.startsWith(SYMBOL_ALL_WORDS_STARTS_WITH)) {
+                    result.add(word);
+                }
+            }
+            Collections.sort(result);
+            return result.toArray(new String[result.size()]);
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FileWork.java:40]:
```suggestion
```

**Pull request closed without merging**